server/sonar-server/src/main/java/org/sonar/server/measure/custom/ws/CustomMeasureJsonWriter.java

```diff
@@ -0,0 +1,99 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.measure.custom.ws;
+
+import java.util.Date;
+import org.sonar.api.measures.Metric;
+import org.sonar.api.server.ServerSide;
+import org.sonar.api.utils.text.JsonWriter;
+import org.sonar.core.component.ComponentDto;
+import org.sonar.core.measure.custom.db.CustomMeasureDto;
+import org.sonar.core.metric.db.MetricDto;
+
+import static org.sonar.server.measure.custom.ws.CreateAction.PARAM_DESCRIPTION;
+import static org.sonar.server.measure.custom.ws.CreateAction.PARAM_PROJECT_ID;
+import static org.sonar.server.measure.custom.ws.CreateAction.PARAM_PROJECT_KEY;
+import static org.sonar.server.measure.custom.ws.CreateAction.PARAM_VALUE;
+
+@ServerSide
+public class CustomMeasureJsonWriter {
+  private static final String FIELD_ID = "id";
+  private static final String FIELD_PROJECT_ID = PARAM_PROJECT_ID;
+  private static final String FIELD_PROJECT_KEY = PARAM_PROJECT_KEY;
+  private static final String FIELD_VALUE = PARAM_VALUE;
+  private static final String FIELD_DESCRIPTION = PARAM_DESCRIPTION;
+  private static final String FIELD_METRIC = "metric";
+  private static final String FIELD_METRIC_KEY = "key";
+  private static final String FIELD_METRIC_ID = "id";
+  private static final String FIELD_METRIC_TYPE = "type";
+  private static final String FIELD_CREATED_AT = "createdAt";
+  private static final String FIELD_UPDATED_AT = "updatedAt";
+
+  public void write(JsonWriter json, CustomMeasureDto measure, MetricDto metric, ComponentDto component) {
+    json.beginObject();
+    json.prop(FIELD_ID, String.valueOf(measure.getId()));
+    json.name(FIELD_METRIC);
+    writeMetric(json, metric);
+    json.prop(FIELD_PROJECT_ID, component.uuid());
+    json.prop(FIELD_PROJECT_KEY, component.key());
+    json.prop(FIELD_DESCRIPTION, measure.getDescription());
+    json.prop(FIELD_VALUE, measureValue(measure, metric));
+    json.propDateTime(FIELD_CREATED_AT, new Date(measure.getCreatedAt()));
+    json.propDateTime(FIELD_UPDATED_AT, new Date(measure.getUpdatedAt()));
+    json.endObject();
+  }
+
+  private String measureValue(CustomMeasureDto measure, MetricDto metric) {
+    Metric.ValueType metricType = Metric.ValueType.valueOf(metric.getValueType());
+    Double doubleValue = measure.getValue();
+    String stringValue = measure.getTextValue();
+
+    switch (metricType) {
+      case BOOL:
+        return doubleValue == 1.0d ? "true" : "false";
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Equality tests should not be made with floating point values. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1244)